shm/src/cargo.cpp

```diff
@@ -0,0 +1,40 @@
+#include "cargo.hpp"
+
+Cargo::Cargo()
+    : Cargo("", 0, 0)
+{}
+Cargo::Cargo(const std::string& name, size_t amount, size_t basePrice)
+    : name_(name)
+    , amount_(amount)
+    , basePrice_(basePrice)
+{}
+
+bool Cargo::operator==(const Cargo& cargo) const {
+    return (name_ == cargo.name_) &&
+           (amount_ == cargo.amount_) &&
+           (basePrice_ == cargo.basePrice_);
+}
+bool Cargo::operator!=(const Cargo& cargo) const {
+    return (name_ != cargo.name_) ||
+           (amount_ != cargo.amount_) ||
+           (basePrice_ != cargo.basePrice_);
+}
```

Comment:
Better return !(*this == cargo) and you will use `operator==`. Please don't duplicate code if not needed :)